src/WebJobs.Script/RuntimeInformation/SystemRuntimeInformation.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class SystemRuntimeInformation : ISystemRuntimeInformation
+    {
+        public Architecture GetOSArchitecture()
+        {
+            return RuntimeInformation.OSArchitecture;
+        }
+
+        public OSPlatform GetOSPlatform()
+        {
```

Comment:
Add these helpers to EnvironmentExtensions. For testing, you can use TestEnvironment

Reply:
This makes testing trickier because I won't be able to call the mocked methods from the `TestEnvironment` unless I also declare them in `IEnvironment` and implement them in `SystemEnvironment`. Adding them to `EnvironmentExtensions` alone does not expose them through `IEnvironment` (which is what I would be passing to the `WorkerConfigFactory` constructor from the DI container). I'm not sure we won't to expose these methods in the `IEnvironment` interface necessarily. Would prefer to keep this as a separate abstraction.

Reply:
These helpers belong in EnvironmentExtensions. Did you try passing IEnvironment as dependency? And for tests you need to TestEnvironment

Reply:
Yes, I tried it locally. When I use `TestEnvironment` without declaring the helpers in `IEnvironment`, I get the static version of the helpers in `EnvironmentExtensions` (i.e. the actual system details) unless I also add them to `IEnvironment` and then define them in `SystemEnvironment`.

Reply:
Lets discuss offline about this.